python/tests/test_tables.py

```diff
@@ -1031,52 +1056,34 @@ def test_set_columns_metadata_schema(self):
 
 
 class AssertEqualsMixin:
-    @pytest.fixture
-    def test_rows(self, scope="session"):
-        test_rows = self.make_transposed_input_data(10)
-        # Annoyingly we have to tweak some types as once added to a row and then put in
-        # an error message things come out differently
-        for n in range(10):
-            for col in test_rows[n].keys():
-                if col in ["timestamp", "record", "ancestral_state", "derived_state"]:
-                    test_rows[n][col] = bytes(test_rows[n][col]).decode("ascii")
-        return test_rows
-
-    @pytest.fixture
-    def table1(self, test_rows):
-        table1 = self.table_class()
-        for row in test_rows[:5]:
-            table1.add_row(**row)
-        return table1
-
-    def test_equal(self, table1, test_rows):
+    def test_equal(self, table_5row, test_rows):
         table2 = self.table_class()
         for row in test_rows[:5]:
             table2.add_row(**row)
-        table1.assert_equals(table2)
+        table_5row.assert_equals(table2)
 
-    def test_type(self, table1):
+    def test_type(self, table_5row):
         with pytest.raises(
             AssertionError,
-            match=f"Types differ: self={type(table1)} other=<class 'int'>",
+            match=f"Types differ: self={type(table_5row)} other=<class 'int'>",
         ):
-            table1.assert_equals(42)
+            table_5row.assert_equals(42)
 
-    def test_metadata_schema(self, table1):
-        if hasattr(table1, "metadata_schema"):
-            assert table1.metadata_schema == tskit.MetadataSchema(None)
-            table2 = table1.copy()
+    def test_metadata_schema(self, table_5row):
```

Comment:
I like the abstract idea of fixtures (i.e., not making up test data over and over again), but I have to say I find this idiom of changing all the variables in a test to (effectively) a global variable really nasty. These tests are now objectively less readable that they were before. Wouldn't a parametrize with a set of locally-defined example tables be better?

Reply:
Hmm, I see, you've made it a 5row table so that you know in later tests that you can do things like ``table[:4]`` or whatever. Maybe disregard above comment then.

Reply:
The main change is moving the fixture to where it can be used by `AssertEqualsMixin` and `FancyIndexingMixin`. I agree the non-locality of fixtures can be a readability issue, but generally think it is worth the trade off.